Support 'more than' and 'less than' statements in verify and try to verify is

peterparser opened the pull request:
This PR adds the support for `more than` and `less than` keywords in `verify` and `try to verify is` statements.

When I was implementing some test cases, I found that this two statements could be useful instead of parsing the data from the kubectl output and do the comparison in a bash fashion.

In case some other changes are needed, please, let me know.

vincent-zurczak commented on lib/detik.bash:
```diff
@@ -168,7 +168,22 @@ verify() {
+			else
+				echo "Found $result $resource named $name (instead of less than $card expected)."
+				return 3
+					
```
Why this empty line here?

vincent-zurczak commented on lib/utils.bash:
```diff
@@ -3,13 +3,17 @@
 # The regex for the "verify" key word
 verify_regex_count_is="^there +is +(0|1) +([a-z]+) +named +'([^']+)'$"
 verify_regex_count_are="^there +are +([0-9]+) +([a-z]+) +named +'([^']+)'$"
+verify_regex_count_less_than="^there are less than +([0-9]+) +([a-z]+) +named +'([^']+)'$"
```
Would be better named `verify_regex_count_is_less_than`.

vincent-zurczak commented on lib/utils.bash:
```diff
@@ -3,13 +3,17 @@
 verify_regex_count_is="^there +is +(0|1) +([a-z]+) +named +'([^']+)'$"
 verify_regex_count_are="^there +are +([0-9]+) +([a-z]+) +named +'([^']+)'$"
+verify_regex_count_less_than="^there are less than +([0-9]+) +([a-z]+) +named +'([^']+)'$"
+verify_regex_count_more_than="^there are more than +([0-9]+) +([a-z]+) +named +'([^']+)'$"
```
Would be better named `verify_regex_count_is_more_than`.

vincent-zurczak commented on tests/test.detik.try.to.verify.is.bats:
```diff
@@ -224,3 +228,41 @@ mytest_with_spaces() {
+	[ ${#lines[@]} -eq 2 ]
+	[ "${lines[0]}" = "Valid expression. Verification in progress..." ]
+	[ "${lines[1]}" = "Current value for nginx is not less than 1..." ]
+}
```
Please, add an empty line at the end of the file.

vincent-zurczak reviewed (changes requested):
Hi @peterparser!

This is a great contribution.
Thank you!

I added few comments, mostly cosmetic.

massimoschembri commented:
> Hi @peterparser!
> 
> This is a great contribution. Thank you!
> 
> I added few comments, mostly cosmetic.

Thanks for the review! I've made the fixes and also made the `less/more than` tests more robust.

peterparser commented:
> Hi @peterparser!
> 
> This is a great contribution. Thank you!
> 
> I added few comments, mostly cosmetic.

Thanks @vincent-zurczak! I've made the changes, also I changed the tests from testing if `more` is present I now check for `more than`, the same happens with `less`. Thanks again

brokenpip3 commented:
hei @peterparser since you are doing this PR can you also add a simple gh action that will run the bats tests? both vs main and during a PR, thanks!

(hoping that is ok for you @vincent-zurczak )

vincent-zurczak commented:
Again, thank you for this contribution.  
I merged it and I was about to create a new tag when I noticed the README documentation had not been updated. Do you think you could do that please?

peterparser commented:
> @peterparser Again, thank you for this contribution. I merged it and I was about to create a new tag when I noticed the README documentation had not been updated. Do you think you could do that please?

Sure, I will update it and put some examples in the `examples` folder!
